### ISSUE TITLE
2021-9-24

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#612]
441What does [0] stand for in input[0]? The first letter of input name; (2021-9-18)
442Replace top_n with top_value; done; replace top_i with top_index; done; (2021-9-18)
443Replace TORCH.TENSOR.TOPK with TORCH.TOPK; done; (2021-9-18)
444Replace category_i with category_index; done; (2021-9-18)
445AG; (2021-9-18)
446Training; Preparing for Training; Before going into training we should make a few helper functions. The first is to interpret the output of the network, which we know to be a likelihood of each category. We can use Tensor.topk to get the index of the greatest value; We will also want a quick way to get a training example (a name and its language); (2021-9-18)
447Replace category_lines with category_names; done; replace func_line with func_name; 
448Training the Network; Now all it takes to train this network is show it a bunch of examples, have it make guesses, and tell it if it’s wrong. For the loss function nn.NLLLoss is appropriate, since the last layer of the RNN is nn.LogSoftmax; (2021-9-18)
449Each loop of training will: (1) Create input and target tensors; (2) Create a zeroed initial hidden state; (3) Read each letter in and; (4) Keep hidden state for next letter; (5) Compare final output to target; (6) Back-propagate; (7) Return the output and loss; (2021-9-18)
